.github/workflows/run-test.yml

```diff
@@ -26,19 +27,7 @@ jobs:
           git tag -a latest -m 'Retag latest commit'
           git push origin latest
 
-  test-action:
-    needs: [add-tag]
-    runs-on: ubuntu-latest
-    steps:
-    - uses: convictional/trigger-workflow-and-wait@v1.6.3
-      with:
-        owner: cpp-linter
-        repo: test-cpp-linter-action
-        github_token: ${{ secrets.PAT_TOKEN }}
-        workflow_file_name: cpp-lint-action.yml
-        ref: master
-        wait_interval: 10
-        client_payload: '{}'
-        propagate_failure: true
-        trigger_workflow: true
-        wait_workflow: true
+  call-test-action:
+    uses: cpp-linter/test-cpp-linter-action/.github/workflows/cpp-lint-action.yml@master
```

Comment:
from the workflow you're calling:
```yml
uses: cpp-linter/cpp-linter-action@latest
```

The way I see it the action repo must pass a ref to the reusable workflow. Then the reusable workflow must then generate a workflow yml (from a template) that incorporates:
```yml
uses: cpp-linter/cpp-linter-action@{{ passed_ref }}
```
and call the generated workflow file (from the reusable workflow - they can be nested). Note, the curly brackets above are for the jinja template syntax (not github workflow syntax)

Reply:
> Do we need to pass a ref from the action repo

The whole point is to use the test repo to dry test the current changes in the action repo.

I've been doing some exploring with reusable workflows (& I ❤️ them), but I don't think they can be used to workaround the strictness of the step's `uses` option.

It would be nice if we could use the composite workflow as a reusable workflow, but I think the reusable workflows need to be located in `.github/workflows/` path. Maybe we could write a pre-commit-time script that updates a generated reusable workflow from the composite workflow...

Reply:
> do you mean to generate a reusable workflow from [action.yml](https://github.com/cpp-linter/cpp-linter-action/blob/main/action.yml)?

Yes.

Reply:
> I'm not particularly familiar with reusable workflows yet

They are GREAT! 💡 If you want to experiment, you could create a reusable workflow in cpp-linter/.github repo that will handle the python releases' CI in the cpp-linter org.

I have a [PR in the nRF24 org](https://github.com/nRF24/.github/pull/3) that does all the C++ CI in reusable workflows for 6 nRF24 org repos. I even have [one that uses cpp-linter pkg](https://github.com/nRF24/.github/blob/reusable-workflow/.github/workflows/cpp_lint.yaml) according to our org requirements.